cirq/contrib/quirk/cells/arithmetic_cells.py

```diff
@@ -0,0 +1,316 @@
+    yield _arithmetic_gate("^A<=B", 1, lambda x, a, b: x ^ int(a <= b))
+    yield _arithmetic_gate("^A>=B", 1, lambda x, a, b: x ^ int(a >= b))
+    yield _arithmetic_gate("^A=B", 1, lambda x, a, b: x ^ int(a == b))
+    yield _arithmetic_gate("^A!=B", 1, lambda x, a, b: x ^ int(a != b))
+
+    # Addition.
+    yield from _arithmetic_family("inc", lambda x: x + 1)
+    yield from _arithmetic_family("dec", lambda x: x - 1)
+    yield from _arithmetic_family("+=A", lambda x, a: x + a)
+    yield from _arithmetic_family("-=A", lambda x, a: x - a)
+
+    # Multiply-accumulate.
+    yield from _arithmetic_family("+=AA", lambda x, a: x + a * a)
+    yield from _arithmetic_family("-=AA", lambda x, a: x - a * a)
+    yield from _arithmetic_family("+=AB", lambda x, a, b: x + a * b)
+    yield from _arithmetic_family("-=AB", lambda x, a, b: x - a * b)
+
+    # Misc.
+    yield from _arithmetic_family("+cntA", lambda x, a: x + _popcnt(a))
+    yield from _arithmetic_family("-cntA", lambda x, a: x - _popcnt(a))
+    yield from _arithmetic_family("^=A", lambda x, a: x ^ a)
+    yield from _arithmetic_family(
+        "Flip<A", lambda x, a: a - x - 1 if x < a else x)
+
+    # Multiplication.
+    yield from _arithmetic_family("*A", lambda x, a: x * a if a & 1 else x)
+    yield from _size_dependent_arithmetic_family(
+        "/A", lambda n: lambda x, a: x * _mod_inv_else_1(a, 1 << n))
+
+    # Modular addition.
+    yield from _arithmetic_family("incmodR", lambda x, r: x + 1)
+    yield from _arithmetic_family("decmodR", lambda x, r: x - 1)
+    yield from _arithmetic_family("+AmodR", lambda x, a, r: x + a)
+    yield from _arithmetic_family("-AmodR", lambda x, a, r: x - a)
+
+    # Modular multiply-accumulate.
+    yield from _arithmetic_family("+ABmodR", lambda x, a, b, r: x + a * b)
+    yield from _arithmetic_family("-ABmodR", lambda x, a, b, r: x - a * b)
+
+    # Modular multiply.
+    yield from _arithmetic_family(
+        "*AmodR", lambda x, a, r: x * _invertible_else_1(a, r))
+    yield from _arithmetic_family(
+        "/AmodR", lambda x, a, r: x * _mod_inv_else_1(a, r))
+    yield from _arithmetic_family(
+        "*BToAmodR", lambda x, a, b, r: x * pow(_invertible_else_1(b, r), a, r))
+    yield from _arithmetic_family(
+        "/BToAmodR", lambda x, a, b, r: x * pow(_mod_inv_else_1(b, r), a, r))
+
+
+def _extended_gcd(a: int, b: int) -> Tuple[int, int, int]:
+    if a == 0:
+        return b, 0, 1
+    gcd, y, x = _extended_gcd(b % a, a)
+    return gcd, x - (b // a) * y, y
+
+
+def _invertible_else_1(a: int, m: int) -> int:
+    """Returns `a` if `a` has a multiplicative inverse, else 1."""
+    i = _mod_inv_else_1(a, m)
+    return a if i != 1 else i
+
+
+def _mod_inv_else_1(a: int, m: int) -> int:
+    """Returns `a**-1` if `a` has a multiplicative inverse, else 1."""
+    if m == 0:
+        return 1
+    gcd, x, _ = _extended_gcd(a % m, m)
+    if gcd != 1:
+        return 1
+    return x % m
+
+
+def _popcnt(a: int) -> int:
```

Comment:
Optional nit: `_hamming` is a more informative name for this function.

Reply:
Acknowledged, but popcnt is also common terminology.